pymc3/smc/sample_smc.py

```diff
@@ -223,10 +231,25 @@ def sample_smc(
     trace.report.nsteps = nsteps
     trace.report._t_sampling = time.time() - t1
 
+    if compute_convergence_checks or return_inferencedata:
+        ikwargs = dict(model=model)
+        idata = to_inference_data(trace, **ikwargs)
+
+    if compute_convergence_checks:
+        if draws < 100:
+            warnings.warn(
+                "The number of samples is too small to check convergence reliably.",
+                stacklevel=2,
+            )
+        else:
+            trace.report._run_convergence_checks(idata, model)
+    trace.report._log_summary()
+
+    posterior = idata if return_inferencedata else trace
     if save_sim_data:
-        return trace, {modelcontext(model).observed_RVs[0].name: np.array(sim_data)}
+        return posterior, {modelcontext(model).observed_RVs[0].name: np.array(sim_data)}
```

Comment:
We should save `sim_data` as posterior predictive samples. The motivation of having `save_sim_data` is that if the simulator is very expensive, we can save resources by saving the simulated data instead of recomputing it

Reply:
Yeah, but we lost the ability to obtain the simulated samples during sampling with the refactoring of Simulator in #4802. I will still try to figure if it's possible to keep them, in which case I'll add it to posterior predictive.

Reply:
Realized there is no point in changing it here, because I have no way of testing it. Will address in #4802